Address the following error seen when compiling for ARM64
/tmp/ccvSdtE2.s: Assembler messages:
/tmp/ccvSdtE2.s:15: Error: operand 1 must be a floating-point register -- `stp fp,lr,[sp,#-16]!'
/tmp/ccvSdtE2.s:16: Error: operand 1 must be an integer register -- `mov fp,sp'
/tmp/ccvSdtE2.s:20: Error: operand 1 must be a floating-point register -- `ldp fp,lr,[sp],#16'

Fix inspired by:  https://bugs.webkit.org/show_bug.cgi?id=175512

--- push_registers_asm.cc       2022-08-31 17:27:04.874557167 +0000
+++ push_registers_asm.cc       2022-08-31 17:29:00.673778358 +0000
@@ -41,9 +41,9 @@
     // Sign return address.
     "  paciasp                                          \n"
 #endif
-    "  stp fp, lr,   [sp, #-16]!                        \n"
+    "  stp x29, x30,   [sp, #-16]!                        \n"
     // Maintain frame pointer.
-    "  mov fp, sp                                       \n"
+    "  mov x29, sp                                       \n"
     // Pass 1st parameter (x0) unchanged (Stack*).
     // Pass 2nd parameter (x1) unchanged (StackVisitor*).
     // Save 3rd parameter (x2; IterateStackCallback)
@@ -52,7 +52,7 @@
     "  mov x2, sp                                       \n"
     "  blr x7                                           \n"
     // Load return address and frame pointer.
-    "  ldp fp, lr, [sp], #16                            \n"
+    "  ldp x29, x30, [sp], #16                            \n"
 #ifdef V8_ENABLE_CONTROL_FLOW_INTEGRITY
     // Authenticate return address.
     "  autiasp                                          \n"
